### PR TITLE
fix: Add Sentry integration for error capturing 

### PR DESCRIPTION
**Pull request by rajusem**
Integrated Sentry for capturing any error in inference pipeline.

### CODE REVIEW

File: deployment/template-cronjob.yaml [hunk 1]
```diff
@@ -81,6 +81,11 @@ objects:
                   secretKeyRef:
                     key: aws_secret_access_key
                     name: openshift-probable-vulnerability-inference-result-s3
+              - name: SENTRY_DSN
+                valueFrom:
+                  secretKeyRef:
+                    name: osa
```
Comment (arajkumar) [inline deployment/template-cronjob.yaml:87]: @rajusem Can't we use same sentry DSN used by dependency analytics? Is it a resource created through app-interface?

Comment (rajusem) [reply deployment/template-cronjob.yaml:87]: @arajkumar : We wanted to keep separate than CRDA project, for the same we have created OSA Project in sentry and using the same. Yes we created new vault-secrete for the same through app-interface.

**Comment by rajusem** [comment]
[test]

**Comment by rajusem** [comment]
[test]

**Comment by rajusem** [comment]
[test]

File: model_inference_triage_pipeline/run_model_inference.py [hunk 2]
```diff
@@ -9,13 +9,22 @@
...
 from utils.bq_utils import get_bq_data_for_inference
 from utils.storage_utils import write_output_csv
-from utils.api_util import save_data_to_db, report_failures, read_probable_cve_data
+
+# Set Sentry log level and initialize it.
+sentry_logging = LoggingIntegration(level=logging.ERROR, event_level=logging.ERROR)
+sentry_sdk.init(dsn=oc.SENTRY_DSN, integrations=[sentry_logging])
+
+# TODO - Remove below line once we test Sentry working in PROD.
+sentry_sdk.capture_exception(Exception("This is an example of an error message."))
```
Comment (arajkumar) [inline model_inference_triage_pipeline/run_model_inference.py]: Can this be a separate PR? So that we can revert this alone as an atomic change?

Comment (rajusem) [reply model_inference_triage_pipeline/run_model_inference.py]: Done

**Review by arajkumar — APPROVED** [review]
lgtm

**Review by rootAvish — APPROVED** [review]
LGTM
